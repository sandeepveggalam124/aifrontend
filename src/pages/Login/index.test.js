// import React from 'react'
// import Login from './index'
// import renderer from 'react-test-renderer'
// import { fireEvent, render,screen, waitFor,act,debug } from '@testing-library/react'
// import {userEvent} from '@testing-library/user-event'
// import { MemoryRouter as Router } from 'react-router-dom'
// // import Toast from '../../../utils/toast'
// import Register from '../Register'






// describe('Login Component',()=>{
//     // render(<Router><Login/></Router>)
//     // const pageTitle=screen.getByRole('heading',{name:'Login'});
//     // const userMsg=screen.getByText(/Enter your Email or phone number as well as passcode/i,{exact:false});
//     // const emailInput = screen.getByPlaceholderText(/email/i);
//     // const passwordInput = screen.getByPlaceholderText(/password/i);
//     // const validationMessage = screen.getByText('Please Enter Valid Email ID or Phone Number', { exact: false });
//     // const ForgotPasswordLink=screen.getByRole('link',{ name: 'Forgot Password?' });

//     // const LoginButton=screen.getByRole('button',{ name: /login/i });
//     // const RegisterButton=screen.getByRole('button',{ name: /register/i });



//     // it('login page rendering ',()=>{
        
       
//     //     expect(userMsg).toBeInTheDocument();
//     //     expect(emailInput).toBeInTheDocument();
//     //     expect(passwordInput).toBeInTheDocument();
//     //     expect(validationMessage).toBeInTheDocument();
//     //     expect(LoginButton).toBeInTheDocument();
//     //     expect(RegisterButton).toBeInTheDocument();
//     //     expect(ForgotPasswordLink).toBeInTheDocument();

//     // })

//     // it("Filling details and clicking buttons in a login Form",async () => {

//     //     render(<Router><Login/></Router>)

//     //     const email=await waitFor(() => {
//     //         return screen.getByPlaceholderText(/email/i);
//     //         //expect(emailInput).toBeInTheDocument();
//     //       });
         
//     //       console.log(email)
 
//     //       await userEvent.type(email, "hello");
       
//     //   });
//     // });
//     let container
//     beforeEach(()=>{
//         const renderResult =render(
//             <Router>
//                 <Login/>
//             </Router>
//         )
//         container=renderResult.container
//     })


//     it('renders correctly',()=>{
        
//         expect(container).toMatchSnapshot();
//     })

//     // it('toggles password visibility', () => {

//     //    const passwordInput = container.querySelector('input[name="password"]');
//     //    const toggleButton = container.querySelector('[data-testid="toggle-password"]');
    
//     //     fireEvent.click(toggleButton);
    
//     //     expect(passwordInput.type).toBe('text');
//     //   });
      
      
      

      
// })
// // test("Filling details and clicking buttons in a login Form", async() => {
// //     render(
// //         <Router>
// //             <Login/>
// //             <Register/>
// //         </Router>
// //     )
// //         const emailInput=screen.getByPlaceholderText(/email/i)
// //         await userEvent.type(emailInput, "test@gmail.com");
// //         const passwordInput=screen.getByPlaceholderText(/password/i)
// //         await userEvent.type(passwordInput, "hello");

// //         const loginButton=screen.getByRole("button",{name:/login/i})

// //         await userEvent.click(loginButton)

// //         // let successToast;
// //         await waitFor(() => {
// //             //successToast = screen.queryByText('logged in');
// //             expect(screen.getByRole('heading',{name:'Sign Up'})).toBeInTheDocument();


// //         });
// //         //expect(successToast.textContent).toContain('logged in');

// //         expect(emailInput).toHaveValue("test@gmail.com")
// //         expect(passwordInput).toHaveValue("hello")

// // });
