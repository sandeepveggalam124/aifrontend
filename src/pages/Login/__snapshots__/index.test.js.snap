// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div>
  <div
    class="d-flex flex-column justify-content-center align-items-center gap-2"
  >
    <h2
      class="text-center mt-1"
    >
      Login
    </h2>
    <form
      class="d-flex flex-column gap-2"
    >
      <p
        class="fw-medium fs-6 text-center"
      >
        Enter your Email or phone number as well as passcode
      </p>
      <div
        class=""
      >
        <input
          autocomplete="off"
          class="form-control form-control-sm details placeholder-custom"
          name="email"
          placeholder="Email ID/PhoneNumber"
          type="email"
        />
      </div>
      <div
        class="input-group input-group-sm"
      >
        <input
          class="form-control form-control-sm details placeholder-custom"
          name="password"
          placeholder="Password*"
          type="password"
        />
        <span
          class="input-group-text details placeholder-custom"
        >
          <i
            class="bi bi-eye eye"
          />
        </span>
      </div>
      <div
        class="form-text fw-bold text-start login-form-text"
      >
        Please Enter Valid Email ID or Phone Number
      </div>
      <p
        class="text-end"
      >
        <a
          class="text-decoration-none forget-pw"
          href="/forgotPassword"
        >
          Forgot Password?
        </a>
      </p>
      <button
        class="btn btn-dark rounded-4 fs-6 fw-bold login-text"
        type="submit"
      >
        Login
      </button>
      <button
        class="btn btn-dark rounded-4 fs-6 fw-bold register-text"
      >
        Register Now
      </button>
    </form>
  </div>
</div>
`;
